src/configuration.ts

```diff
@@ -128,15 +129,46 @@ export function loadConfigurationFromPath(configFilePath: string): IConfiguratio
         console.warn(PACKAGE_DEPRECATION_MSG);
         return require(configFilePath).tslintConfig;
     } else {
-        let fileData = fs.readFileSync(configFilePath, "utf8");
-        // remove BOM if present
-        fileData = fileData.replace(/^\uFEFF/, "");
-        const configFile: IConfigurationFile = JSON.parse(fileData);
-        configFile.rulesDirectory = getRulesDirectories(configFile.rulesDirectory, path.dirname(configFilePath));
+        let configFile: IConfigurationFile;
+        let configFileDir: string;
+        if (fs.existsSync(configFilePath)) {
+            let fileData = fs.readFileSync(configFilePath, "utf8");
+            // remove BOM if present
+            fileData = fileData.replace(/^\uFEFF/, "");
+            configFile = JSON.parse(fileData);
+            configFileDir = path.dirname(configFilePath);
+        } else {
+            configFile = require(configFilePath);
+            configFileDir = path.resolve(`./node_modules/${configFilePath}`);
+        }
+
+        configFile.rulesDirectory = getRulesDirectories(configFile.rulesDirectory, configFileDir);
+        configFile.extends = arrayify(configFile.extends);
+
+        for (const name of configFile.extends) {
+            const baseConfigFilePath = resolveConfigurationPath(name, configFileDir);
+            const baseConfigFile = loadConfigurationFromPath(baseConfigFilePath);
+            configFile = extendConfigurationFile(configFile, baseConfigFile);
+        }
         return configFile;
     }
 }
 
+/**
+ * Resolve configuration file path
+ * @var relativeFilePath Relative path or package name (tslint-config-X) or package short name (X)
+ */
+function resolveConfigurationPath(relativeFilePath: string, relativeTo: string) {
```

Comment:
I think this function should have three cases (pseduocode below):

```
if path.isAbsolute(filePath):  // it could be an absolute path
  return filePath
else if filePath.startsWith("."):
  return getRelativePath(filePath, relativeTo)  // ha, this should really be called resolveRelativePath, but too late now
else:
  return require.resolve(filePath);
```

So no matter what, it returns an absolute path to a file. Also, let's not do any magic prefixing by adding "tslint-config-": if they provide a package name let's just use it as is.